src/api/actions/helpers.ts

```diff
@@ -28,8 +29,14 @@ export const makeApiCall = async <T>(
     throw Error(`Request failed with code ${res.status}: ${await res.text()}`);
   }
 
-  const returnedBody = await res.json();
-  return returnedBody as T;
+  if (softFail) {
+    const returnedBody = await res.json();
+    returnedBody["softFailMessage"] = softFailMessage;
```

Comment:
Interesting solution. This might fail in the case that the error response is not JSON, but hopefully it will be.